--- Code/AnimationExportLib.py	2025-09-11 10:08:01.913601819 +0800
+++ ../Assembly4-0.50.19/AnimationExportLib.py	2025-07-04 10:43:18.148170605 +0800
@@ -518,33 +509,33 @@
         self.outputVLayout.addWidget(self.outputFileSel)
 
         self.outputHL2 = QtGui.QHBoxLayout()
-        self.outputHL2.addWidget(QtGui.QLabel("Width:"))
+        self.outputHL2.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_Animate", "Width:")))
         self.outputHL2.addWidget(self.sbOutWidth)
-        self.outputHL2.addWidget(QtGui.QLabel("Height:"))
+        self.outputHL2.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_Animate", "Height:")))
         self.outputHL2.addWidget(self.sbOutHeight)
         self.outputVLayout.addLayout(self.outputHL2)
 
         self.outputHL3 = QtGui.QHBoxLayout()
-        self.outputHL3.addWidget(QtGui.QLabel("FPS:"))
+        self.outputHL3.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_Animate", "FPS:")))
         self.outputHL3.addWidget(self.sbOutFPS)
-        self.outputHL3.addWidget(QtGui.QLabel("Loops:"))
+        self.outputHL3.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_Animate", "Loops:")))
         self.outputHL3.addWidget(self.sbOutLoops)
         self.outputVLayout.addLayout(self.outputHL3)
 
         self.outputHL4 = QtGui.QHBoxLayout()
-        self.outputHL4.addWidget(QtGui.QLabel("Smoothen:"))
+        self.outputHL4.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_Animate", "Smoothen:")))
         self.outputHL4.addWidget(self.sbSmoothFactor)
         self.outputVLayout.addLayout(self.outputHL4)
 
         self.gpbOutput.setLayout(self.outputVLayout)
 
         # # # Background Group Box # # #
-        self.gpbBG = QtGui.QGroupBox("Background", self)
+        self.gpbBG = QtGui.QGroupBox(App.Qt.translate("Asm4_Animate", "Background"), self)
         self.gpbBG.setCheckable(True)
         self.gpbBG.setChecked(False)
         self.bgImgFileSel = fileSelectorWidget("read", self.gpbBG)
-        tt = 'Choose an image file\n'
-        tt+= 'Supported extensions are *.png *.jpg *.jpeg *.gif'
+        tt = App.Qt.translate("Asm4_Animate", 'Choose an image file\n'
+        + 'Supported extensions are *.png *.jpg *.jpeg *.gif')
         self.bgImgFileSel.setToolTip(tt)
         self.bgColorSel = colorSelectorWidget((255, 255, 255, 255), self.gpbBG)
 
@@ -555,11 +546,11 @@
         self.gpbBG.setLayout(self.bgVLayout)
 
         # # # Logo Group Box # # #
-        self.gpbLogo = QtGui.QGroupBox("Logo", self)
+        self.gpbLogo = QtGui.QGroupBox(App.Qt.translate("Asm4_Animate", "Logo"), self)
         self.logoFileSel = fileSelectorWidget("read", self.gpbLogo)
         self.logoFileSel.setFile(os.path.join(Asm4.iconPath, 'FreeCad.png'))
-        tt = 'Choose an image file\n'
-        tt+= 'Supported extensions are *.png *.jpg *.jpeg *.gif'
+        tt = App.Qt.translate("Asm4_Animate", 'Choose an image file\n'
+        + 'Supported extensions are *.png *.jpg *.jpeg *.gif')
         self.logoFileSel.setToolTip(tt)
 
         self.sbLogoWidth = QtGui.QSpinBox()
@@ -588,16 +579,16 @@
         self.LogoVLayout.addWidget(self.logoFileSel)
 
         self.LogoHL2 = QtGui.QHBoxLayout()
-        self.LogoHL2.addWidget(QtGui.QLabel("Width:"))
+        self.LogoHL2.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_Animate", "Width:")))
         self.LogoHL2.addWidget(self.sbLogoWidth)
-        self.LogoHL2.addWidget(QtGui.QLabel("Height:"))
+        self.LogoHL2.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_Animate", "Height:")))
         self.LogoHL2.addWidget(self.sbLogoHeight)
         self.LogoVLayout.addLayout(self.LogoHL2)
 
         self.LogoHL3 = QtGui.QHBoxLayout()
-        self.LogoHL3.addWidget(QtGui.QLabel("X-Offset:"))
+        self.LogoHL3.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_Animate", "X-Offset:")))
         self.LogoHL3.addWidget(self.sbLogoX)
-        self.LogoHL3.addWidget(QtGui.QLabel("Y-Offset:"))
+        self.LogoHL3.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_Animate", "Y-Offset:")))
         self.LogoHL3.addWidget(self.sbLogoY)
         self.LogoVLayout.addLayout(self.LogoHL3)
 
@@ -617,10 +608,10 @@
         self.upperHLayout.addLayout(self.upperRHVLayout)
 
     # # # buttons incl. lower layout # # #
-        self.pbClose = QtGui.QPushButton("Close")
-        self.pbRefreshPreview = QtGui.QPushButton("Refresh Preview")
-        self.pbCreateAndSave = QtGui.QPushButton("Create and Save")
-        self.pbDummy = QtGui.QPushButton("Dummy")
+        self.pbClose = QtGui.QPushButton(App.Qt.translate("Asm4_Animate", "Close"))
+        self.pbRefreshPreview = QtGui.QPushButton(App.Qt.translate("Asm4_Animate", "Refresh Preview"))
+        self.pbCreateAndSave = QtGui.QPushButton(App.Qt.translate("Asm4_Animate", "Create and Save"))
+        self.pbDummy = QtGui.QPushButton(App.Qt.translate("Asm4_Animate", "Dummy"))
         self.pbDummy.setDefault(True)
         self.pbDummy.setVisible(False)
 
@@ -680,9 +671,9 @@
     def __init__(self, type="read", parent=None):
         super().__init__(parent)
 
-        self.label = QtGui.QLabel("File:", parent)
+        self.label = QtGui.QLabel(App.Qt.translate("Asm4_Animate", "File:"), parent)
         self.leFilename = QtGui.QLineEdit(parent)
-        self.pbSelectFile = QtGui.QPushButton("Select", parent)
+        self.pbSelectFile = QtGui.QPushButton(App.Qt.translate("Asm4_Animate", "Select"), parent)
         #self.pbSelectFile.resize(50, self.pbSelectFile.height())
         #self.pbSelectFile.setFixedWidth(50)
 
@@ -693,8 +684,8 @@
         self.setLayout(self.lowerButtonHLayout)
 
         self.type = type
-        self.title = "Select File"
-        self.filter = "Image Files (*.png *.jpg *.jpeg *.gif)" if self.type=="read" else "Supported Files (*.mp4 *.avi *.mov *.mkv *.gif *.png)"
+        self.title = App.Qt.translate("Asm4_Animate", "Select File")
+        self.filter = App.Qt.translate("Asm4_Animate", "Image Files (*.png *.jpg *.jpeg *.gif)") if self.type=="read" else App.Qt.translate("Asm4_Animate", "Supported Files (*.mp4 *.avi *.mov *.mkv *.gif *.png)")
 
         self.pbSelectFile.clicked.connect(self.selectFile)
 
@@ -734,10 +725,10 @@
     def __init__(self, initialColor=(255, 255, 255, 255), parent=None):
         super().__init__(parent)
 
-        self.label = QtGui.QLabel("Color:", parent)
+        self.label = QtGui.QLabel(App.Qt.translate("Asm4_Animate", "Color:"), parent)
         self.leColor = QtGui.QLineEdit(parent)
         self.leColor.setReadOnly(True)
-        self.pbSelect = QtGui.QPushButton("Select", parent)
+        self.pbSelect = QtGui.QPushButton(App.Qt.translate("Asm4_Animate", "Select"), parent)
         #self.pbSelect.resize(50, self.pbSelect.height())
         #self.pbSelect.setFixedWidth(50)
 
