--- Code/Asm4_Measure.py	2025-09-11 10:24:58.403581823 +0800
+++ ../Assembly4-0.50.19/Asm4_Measure.py	2025-07-04 11:07:53.896220187 +0800
@@ -102,8 +93,8 @@
         super(MeasureCmd,self).__init__()
 
     def GetResources(self):
-        return {"MenuText": "Measure",
-                "ToolTip": "Measure Tool",
+        return {"MenuText": App.Qt.translate("Asm4", "Measure"),
+                "ToolTip": App.Qt.translate("Asm4", "Measure Tool"),
                 "Pixmap" : os.path.join( iconDir , 'Part_Measure.svg')
                 }
 
@@ -130,7 +121,7 @@
         self.form = self.base
         iconFile = os.path.join( iconDir , 'Part_Measure.svg')
         self.form.setWindowIcon(QtGui.QIcon( iconFile ))
-        self.form.setWindowTitle('Measure')
+        self.form.setWindowTitle(App.Qt.translate("Asm4_Measure",'Measure'))
 
         # remove selectionFilter
         global Asm4_3DselObserver
@@ -150,7 +141,7 @@
         #self.so=SelObserverCaliper()
         self.so = selectionObserver()
         Gui.Selection.addObserver( self.so, 1 ) # 1 = resolve
-        FCC.PrintMessage('Observer started\n')
+        FCC.PrintMessage(App.Qt.translate("Asm4_Measure", 'Observer started\n'))
 
         # enable the measurement points
         self.Selection1.setEnabled(True)
@@ -182,12 +173,12 @@
 
     # Close
     def Finish(self):
-        FCC.PrintMessage("closing ... ")
+        FCC.PrintMessage(App.Qt.translate("Asm4_Measure", "closing ... "))
         try:
             Gui.Selection.removeObserver(self.so)   # uninstall the resident SelObserver function
-            FCC.PrintMessage("done\n")
+            FCC.PrintMessage(App.Qt.translate("Asm4_Measure", "done\n"))
         except:
-            FCC.PrintWarning("was not able to remove observer\n")
+            FCC.PrintWarning(App.Qt.translate("Asm4_Measure", "was not able to remove observer\n"))
         # remove PtS because it can have strange results
         removePtS()
         # close Task widget
@@ -198,7 +189,7 @@
         global PtS, addedDims
         Gui.Selection.clearSelection()
         self.clearConsole()
-        FCC.PrintMessage('Removing all measurements ...')
+        FCC.PrintMessage(App.Qt.translate("Asm4_Measure", 'Removing all measurements ...'))
         removePtS()
         for d in addedDims:
             FCC.PrintMessage('.')
@@ -220,7 +211,7 @@
         self.Selection1.setChecked(False)
         self.Selection2.setEnabled(False)
         self.resultText.clear()
-        FCC.PrintMessage(' done\n')
+        FCC.PrintMessage(App.Qt.translate("Asm4_Measure", ' done\n'))
 
 
     # clear report view and Python panel
@@ -229,7 +220,7 @@
         mw = Gui.getMainWindow()
         #c=mw.findChild(QtGui.QPlainTextEdit, "Python console")
         #c.clear()
-        rv = mw.findChild(QtGui.QTextEdit, "Report view")
+        rv = mw.findChild(QtGui.QTextEdit, App.Qt.translate("Asm4_Measure","Report view"))
         rv.clear()
 
     # Actions
@@ -294,7 +285,7 @@
 
         # the layout for the main window is vertical (top to down)
         self.mainLayout = QtGui.QVBoxLayout(self.form)
-        self.mainLayout.addWidget(QtGui.QLabel('Controls'))
+        self.mainLayout.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_Measure", 'Controls')))
 
         # measurement type
         self.measureGroup = QtGui.QFrame(self.form)
@@ -307,7 +298,7 @@
         pm.loadFromData(base64.b64decode(Dim_Radius_b64))
         self.rbRadius = QtGui.QRadioButton(self.measureGroup)
         self.rbRadius.setObjectName("rbRadius")
-        self.rbRadius.setToolTip("Measure Radius of Arc or Circle\nMeasure Length of Edge")
+        self.rbRadius.setToolTip(App.Qt.translate("Asm4_Measure", "Measure Radius of Arc or Circle\nMeasure Length of Edge"))
         self.rbRadius.setIconSize(QtCore.QSize(btn_md_sizeX,btn_md_sizeY))
         self.rbRadius.setIcon(QtGui.QIcon(pm))
         self.rbRadius.setChecked(True)
@@ -316,7 +307,7 @@
         pm.loadFromData(base64.b64decode(Dim_Length_b64))
         self.rbDistance = QtGui.QRadioButton(self.measureGroup)
         self.rbDistance.setObjectName("rbDistance")
-        self.rbDistance.setToolTip("Measure Distance")
+        self.rbDistance.setToolTip(App.Qt.translate("Asm4_Measure", "Measure Distance"))
         self.rbDistance.setIconSize(QtCore.QSize(btn_md_sizeX,btn_md_sizeY))
         self.rbDistance.setIcon(QtGui.QIcon(pm))
         self.measureGrid.addWidget(self.rbDistance, 0, 1 )
@@ -324,7 +315,7 @@
         pm.loadFromData(base64.b64decode(Dim_Angle_b64))
         self.rbAngle = QtGui.QRadioButton(self.measureGroup)
         self.rbAngle.setObjectName("rbAngle")
-        self.rbAngle.setToolTip("Measure Angle")
+        self.rbAngle.setToolTip(App.Qt.translate("Asm4_Measure","Measure Angle"))
         self.rbAngle.setIconSize(QtCore.QSize(btn_md_sizeX,btn_md_sizeY))
         self.rbAngle.setIcon(QtGui.QIcon(pm))
         self.measureGrid.addWidget(self.rbAngle, 0, 2 )
@@ -342,7 +333,7 @@
         pm.loadFromData(base64.b64decode(Snap_Options_b64))
         self.rbSnap = QtGui.QRadioButton(self.snapGroup)
         self.rbSnap.setObjectName("rbSnap")
-        self.rbSnap.setToolTip("Snap to EndPoint, MiddlePoint, Center")
+        self.rbSnap.setToolTip(App.Qt.translate("Asm4_Measure","Snap to EndPoint, MiddlePoint, Center"))
         self.rbSnap.setIconSize(QtCore.QSize(3*btn_md_sizeX,btn_md_sizeY))
         self.rbSnap.setIcon(QtGui.QIcon(pm))
         self.rbSnap.setChecked(False)
@@ -352,7 +343,7 @@
         pm.loadFromData(base64.b64decode(Center_Mass_b64))
         self.rbShape = QtGui.QRadioButton(self.snapGroup)
         self.rbShape.setObjectName("rbShape")
-        self.rbShape.setToolTip("Select Shape")
+        self.rbShape.setToolTip(App.Qt.translate("Asm4_Measure","Select Shape"))
         self.rbShape.setIconSize(QtCore.QSize(btn_md_sizeX,btn_md_sizeY))
         self.rbShape.setIcon(QtGui.QIcon(pm))
         self.rbShape.setChecked(True)
@@ -433,8 +424,8 @@
         # select elements
         self.selectGrid = QtGui.QGridLayout()
         # first element
-        self.Selection1 = QtGui.QPushButton('Selection 1')
-        self.Selection1.setToolTip("Select First Element")
+        self.Selection1 = QtGui.QPushButton(App.Qt.translate("Asm4_Measure",'Selection 1'))
+        self.Selection1.setToolTip(App.Qt.translate("Asm4_Measure","Select First Element"))
         self.Selection1.setMaximumWidth(150)
         self.Selection1.setCheckable(True)
         self.Selection1.setChecked(False)
@@ -452,8 +443,8 @@
         self.selectGrid.addWidget(self.sel1Icon,   0,2)
 
         # second element
-        self.Selection2 = QtGui.QPushButton('Selection 2')
-        self.Selection2.setToolTip("Select Second Element")
+        self.Selection2 = QtGui.QPushButton(App.Qt.translate("Asm4_Measure",'Selection 2'))
+        self.Selection2.setToolTip(App.Qt.translate("Asm4_Measure","Select Second Element"))
         self.Selection2.setMaximumWidth(150)
         self.Selection2.setEnabled(False)
         self.Selection2.setChecked(False)
@@ -476,21 +467,21 @@
         # draw annotation in the GUI window
         self.bLabel = QtGui.QCheckBox()
         self.bLabel.setObjectName("bLabel")
-        self.bLabel.setToolTip("Enable extra Label")
-        self.bLabel.setText("Show Label in 3D view")
+        self.bLabel.setToolTip(App.Qt.translate("Asm4_Measure", "Enable extra Label"))
+        self.bLabel.setText(App.Qt.translate("Asm4_Measure", "Show Label in 3D view"))
         self.bLabel.setChecked(True)
         self.mainLayout.addWidget(self.bLabel)
 
         # draw X-Y-Z components
         self.Components = QtGui.QCheckBox()
         self.Components.setObjectName("Components")
-        self.Components.setToolTip("Show all dimension components")
-        self.Components.setText("Show Components")
+        self.Components.setToolTip(App.Qt.translate("Asm4_Measure","Show all dimension components"))
+        self.Components.setText(App.Qt.translate("Asm4_Measure","Show Components"))
         self.Components.setChecked(False)
         self.mainLayout.addWidget(self.Components)
 
         # Results
-        self.mainLayout.addWidget(QtGui.QLabel('Results'))
+        self.mainLayout.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_Measure",'Results')))
         self.resultText = QtGui.QTextEdit()
         self.resultText.setMinimumSize(200, 200)
         self.resultText.setReadOnly(True)
@@ -657,10 +648,10 @@
                                     self.measureLine( self.Shp1 )
                                 # dunno what that stuff is
                                 else:
-                                    self.printResult("Can't measure\n"+str(self.Shp1))
+                                    self.printResult(App.Qt.translate("Asm4_Measure","Can't measure\n")+str(self.Shp1))
                             # dunno what that stuff is
                             else:
-                                self.printResult("Can't measure\n"+str(subShape))
+                                self.printResult(App.Qt.translate("Asm4_Measure","Can't measure\n")+str(subShape))
                             # unset first selection
                             self.Sel1 is None
                         # if not rbRadius, launch the selection of the second element
@@ -712,7 +703,7 @@
                                 if self.Sel1=='shape' and self.Sel2=='shape':
                                     self.angleShapes( self.Shp1, self.Shp2 )
                                 else:
-                                    self.printResult( 'Select only faces or lines' )
+                                    self.printResult( App.Qt.translate("Asm4_Measure",'Select only faces or lines' ))
                         # some problem
                         else:
                             self.printResult( 'ERROR 44\n'+str(self.Shp2) )
@@ -726,7 +717,7 @@
         global taskUI
         if shape1.isValid() and shape2.isValid():
             Gui.Selection.clearSelection()
-            self.printResult( 'Measuring angles' )
+            self.printResult( App.Qt.translate("Asm4_Measure",'Measuring angles') )
             # Datum object
             if shape1.BoundBox.DiagonalLength > 1e+10:
                 pt1 = shape1.Placement.Base
@@ -764,9 +755,9 @@
                 except:
                     pass
             else:
-                self.printResult('Ivalid directions')
+                self.printResult(App.Qt.translate("Asm4_Measure",'Ivalid directions'))
         else:
-            self.printResult('Ivalid shapes')
+            self.printResult(App.Qt.translate("Asm4_Measure",'Ivalid shapes'))
 
     # uses BRepExtrema_DistShapeShape to calculate the distance between 2 shapes
     def distShapes( self, shape1, shape2 ):
@@ -776,13 +767,13 @@
             measure = shape1.distToShape(shape2)
             if measure and self.isVector(measure[1][0][0]) and self.isVector(measure[1][0][1]):
                 dist = measure[0]
-                self.printResult('Minimum Distance :\n  '+str(dist))
+                self.printResult(App.Qt.translate("Asm4_Measure",'Minimum Distance :\n  ')+str(dist))
                 if dist > 1.0e-9:
                     pt1   = measure[1][0][0]
                     pt2   = measure[1][0][1]
                     self.measurePoints(pt1,pt2)
         else:
-            self.printResult('Ivalid shapes')
+            self.printResult(App.Qt.translate("Asm4_Measure",'Ivalid shapes'))
 
     # measure a straight line
     def measureLine(self, line ):
@@ -796,7 +787,7 @@
             dy = pt1[1]-pt2[1]
             dz = pt1[2]-pt2[2]
             length = line.Length
-            text = 'Length = '+self.render_distance(length)+'\n'
+            text = App.Qt.translate("Asm4_Measure",'Length = ')+self.render_distance(length)+'\n'
             text += "ΔX = "+self.render_distance(dx)+"\n"
             text += 'ΔY = '+self.render_distance(dy)+'\n'
             text += 'ΔZ = '+self.render_distance(dz)
@@ -811,7 +802,7 @@
                     anno = ['L = '+self.render_distance(length)]
                 self.drawAnnotation( mid, anno )
         else:
-            self.printResult( 'Not a valid Line\n'+str(line) )
+            self.printResult( App.Qt.translate("Asm4_Measure",'Not a valid Line\n')+str(line) )
 
     # measure distance between 2 points
     def measurePoints(self, pt1, pt2 ):
@@ -824,7 +815,7 @@
             dy = pt1[1]-pt2[1]
             dz = pt1[2]-pt2[2]
             dist = math.sqrt(dx*dx + dy*dy + dz*dz)
-            text = 'Distance = '+self.render_distance(dist)+'\n'
+            text = App.Qt.translate("Asm4_Measure",'Distance = ')+self.render_distance(dist)+'\n'
             text += "ΔX : "+self.render_distance(dx)+"\n"
             text += 'ΔY : '+self.render_distance(dy)+'\n'
             text += 'ΔZ : '+self.render_distance(dz)
@@ -838,7 +829,7 @@
                     anno = ['D = '+self.render_distance(dist)]
                 self.drawAnnotation( mid, anno )
         else:
-            self.printResult( 'Not valid Points' )
+            self.printResult( App.Qt.translate("Asm4_Measure",'Not valid Points') )
 
     # measure radius of a circle
     def measureCircle(self, circle):
@@ -849,12 +840,12 @@
             axis   = circle.Curve.Axis
             Gui.Selection.clearSelection()
             self.drawCircle( radius, center, axis )
-            text = 'Radius : '+self.render_distance(radius)+"\n"
+            text = App.Qt.translate("Asm4_Measure",'Radius : ')+self.render_distance(radius)+"\n"
             # if annotation is checked, show label with R = radius
-            text += "Diameter : "+self.render_distance(radius*2)+"\n"
-            text += 'Center : \n'
+            text += App.Qt.translate("Asm4_Measure","Diameter : ")+self.render_distance(radius*2)+"\n"
+            text += App.Qt.translate("Asm4_Measure",'Center : \n')
             text += '  ( '+self.arrondi(center.x)+", "+self.arrondi(center.y)+", "+self.arrondi(center.z)+" )\n"
-            text += 'Axis : \n'
+            text += App.Qt.translate("Asm4_Measure",'Axis : \n')
             text += "  ( "+self.arrondi(axis.x)+", "+self.arrondi(axis.y)+", "+self.arrondi(axis.z)+" )"
             self.printResult(text)
             if taskUI.bLabel.isChecked():
@@ -864,7 +855,7 @@
             else:
                 PtS = self.drawPoint(center)
         else:
-            self.printResult('Not a valid circle\n'+str(circle))
+            self.printResult(App.Qt.translate("Asm4_Measure",'Not a valid circle\n')+str(circle))
 
 
     # figure out the direction of a shape, be it a line, a surface or a circle
@@ -906,7 +897,7 @@
             elif hasattr(shape,'BoundBox'):
                 point = shape.BoundBox.Center
         else:
-            self.printResult('Invalid shape\n'+str(shape))
+            self.printResult(App.Qt.translate("Asm4_Measure",'Invalid shape\n')+str(shape))
         return point
 
     # measure the coordinates of a single point
@@ -920,11 +911,11 @@
                                         and len(vertex.Vertexes) > 0:
             point = vertex.Vertexes[0].Point
         else:
-            self.printResult('Not a valid point\n'+str(vertex))
+            self.printResult(App.Qt.translate("Asm4_Measure",'Not a valid point\n')+str(vertex))
         if point:
             #self.printResult( 'Measuring coordinates of\n'+str(vertex) )
-            anno = ['Coordinates :', 'X : '+self.arrondi(point.x), 'Y : '+self.arrondi(point.y), 'Z : '+self.arrondi(point.z)]
-            text =  'Coordinates :\n'
+            anno = [App.Qt.translate("Asm4_Measure",'Coordinates :'), 'X : '+self.arrondi(point.x), 'Y : '+self.arrondi(point.y), 'Z : '+self.arrondi(point.z)]
+            text =  App.Qt.translate("Asm4_Measure",'Coordinates :\n')
             text += "X : "+str(point.x)+"\n"
             text += 'Y : '+str(point.y)+'\n'
             text += 'Z : '+str(point.z)
@@ -936,24 +927,24 @@
     def measureArea(self, face ):
         if face.isValid() and hasattr(face,'Area'):
             if self.isFlatFace(face):
-                self.printResult('Flat face\nArea : '+str(face.Area)+'\n')
+                self.printResult(App.Qt.translate("Asm4_Measure",'Flat face\nArea : ')+str(face.Area)+'\n')
             else:
-                self.printResult('Area : '+str(face.Area)+"\n")
+                self.printResult(App.Qt.translate("Asm4_Measure",'Area : ')+str(face.Area)+"\n")
         else:
-            self.printResult('Not a valid surface\n'+str(face) )
+            self.printResult(App.Qt.translate("Asm4_Measure",'Not a valid surface\n')+str(face) )
 
 
     def printDims(self, ds, dx, dy, dz, dimType='Distance'):
-        text = dimType+' : '+str(ds)
+        text = App.Qt.translate("Asm4_Measure",dimType)+' : '+str(ds)
         text += "\nΔX = "+str(dx)+"\nΔY = "+str(dy)+"\nΔZ = "+str(dz)
         self.printResult(text)
 
     def printAngle(self, angle, distance=-1 ):
         global taskUI
         taskUI.resultText.clear()
-        text = 'Angle : '+str(angle)+'°\n'
+        text = App.Qt.translate("Asm4_Measure",'Angle : ')+str(angle)+'°\n'
         if distance != -1:
-            text += 'Distance // '+str(distance)
+            text += App.Qt.translate("Asm4_Measure",'Distance // ')+str(distance)
         taskUI.resultText.setPlainText(text)
 
     # print the result in the text field of the UI
@@ -1045,7 +1036,7 @@
         if distance == -1 or taskUI.Components.isChecked()==False :
             anno.LabelText = [self.arrondi(angle)+'°']
         else:
-            anno.LabelText = ['Angle: '+self.arrondi(angle)+'°', 'Distance // '+self.arrondi(distance)]
+            anno.LabelText = [App.Qt.translate("Asm4_Measure",'Angle: ')+self.arrondi(angle)+'°', App.Qt.translate("Asm4_Measure",'Distance // ')+self.arrondi(distance)]
         annoG = Gui.ActiveDocument.getObject(anno.Name)
         annoG.FontSize = annoFontSize
         self.addToDims(anno)
