--- Code/makeBinderCmd.py	2025-09-08 10:52:28.586123060 +0800
+++ ../Assembly4-0.50.19/makeBinderCmd.py	2025-07-04 14:32:53.732650920 +0800
@@ -30,12 +30,12 @@
         pass
 
     def GetResources(self):
-        tooltip  = "Create a reference to an external shape\n"
-        tooltip += "This creates a SubShapeBinder of the selected shapes\n"
-        tooltip += "(face, edge, point) in the root assembly\n"
-        tooltip += "Only shapes belonging to the same part can be imported in a single step"
+        tooltip  = App.Qt.translate("Asm4_shapeBinder", "Create a reference to an external shape\n"
+        + "This creates a SubShapeBinder of the selected shapes\n"
+        + "(face, edge, point) in the root assembly\n"
+        + "Only shapes belonging to the same part can be imported in a single step")
         iconFile = os.path.join( Asm4.iconPath, 'Asm4_shapeBinder.svg' )
-        return {"MenuText": "Create a shape binder", "ToolTip":  tooltip, "Pixmap": iconFile}
+        return {"MenuText": App.Qt.translate("Asm4_shapeBinder", "Create a shape binder"), "ToolTip":  tooltip, "Pixmap": iconFile}
 
     def IsActive(self):
         # only do this for assembly objects and all selected shapes must be in the same part
