--- Code/placeLinkCmd.py	2025-09-08 10:52:28.587123099 +0800
+++ ../Assembly4-0.50.19/placeLinkCmd.py	2025-07-04 14:52:11.245754169 +0800
@@ -33,8 +33,8 @@
         super(placeLinkCmd,self).__init__()
 
     def GetResources(self):
-        return {"MenuText": "Edit Placement of a Part",
-                "ToolTip": "Move/Attach a Part in the assembly",
+        return {"MenuText": App.Qt.translate("Asm4_placePart", "Edit Placement of a Part"),
+                "ToolTip": App.Qt.translate("Asm4_placePart", "Move/Attach a Part in the assembly"),
                 "Pixmap" : os.path.join( Asm4.iconPath , 'Place_Link.svg')
                 }
 
@@ -66,21 +66,21 @@
                 if Asm4.isAsm4EE(selection):
                     # BUGFIX: if the part was corrupted by Assembly4 v0.11.5:
                     if hasattr(selection,'MapMode'):
-                        Asm4.warningBox("This Part has the Attachment extension, it can only be placed manually")
+                        Asm4.warningBox(App.Qt.translate("Asm4_placePart", "This Part has the Attachment extension, it can only be placed manually"))
                     else:
                         # launch the UI in the task panel
                         ui = placeLinkUI()
                         Gui.Control.showDialog(ui)
                 # else try to convert it
                 else:
-                    convert = Asm4.confirmBox("This Part wasn't assembled with this Assembly4 WorkBench, but I can convert it.")
+                    convert = Asm4.confirmBox(App.Qt.translate("Asm4_placePart", "This Part wasn't assembled with this Assembly4 WorkBench, but I can convert it."))
                     if convert:
                         Asm4.makeAsmProperties( selection, reset=True )
                         # launch the UI in the task panel
                         ui = placeLinkUI()
                         Gui.Control.showDialog(ui)
             else:
-                Asm4.warningBox('Please select a link in the assembly Model.')
+                Asm4.warningBox(App.Qt.translate("Asm4_placePart", 'Please select a link in the assembly Model.'))
         else:
             # or any part that has a Placement ?
             if len(Gui.Selection.getSelection())==1:
@@ -105,7 +105,7 @@
                                 Gui.Control.showDialog(ui)
                             # else try to convert it
                             else:
-                                convert = Asm4.confirmBox("This Part wasn't assembled with this Assembly4 WorkBench, but I can convert it.")
+                                convert = Asm4.confirmBox(App.Qt.translate("Asm4_placePart", "This Part wasn't assembled with this Assembly4 WorkBench, but I can convert it."))
                                 if convert:
                                     Asm4.makeAsmProperties( selection, reset=True )
                                     # launch the UI in the task panel
@@ -113,7 +113,7 @@
                                     Gui.Control.showDialog(ui)
                         # the selected object doesn't belong to the root assembly
                         else:
-                            Asm4.warningBox('Please select an object in the assembly Model.')
+                            Asm4.warningBox(App.Qt.translate("Asm4_placePart", 'Please select an object in the assembly Model.'))
                             return
 
     
