--- Code/placeLinkUI.py	2025-09-11 10:38:23.063223438 +0800
+++ ../Assembly4-0.50.19/placeLinkUI.py	2025-07-04 16:48:41.646969936 +0800
@@ -75,7 +67,7 @@
             self.old_linkLCS = self.selectedObj.AttachedBy[1:]
         else:
             # this shouldn't happen
-            FCC.PrintWarning("WARNING : unsupported Assembly/Solver/Part combination, you shouldn't be seeing this\n")
+            FCC.PrintWarning(App.Qt.translate("Asm4_placePart", "WARNING : unsupported Assembly/Solver/Part combination, you shouldn't be seeing this\n"))
             Asm4.makeAsmProperties(self.selectedObj)
             self.old_AO = []
             self.old_linkLCS = ''
@@ -87,7 +79,7 @@
         self.form = QtGui.QWidget()
         iconFile = os.path.join( Asm4.iconPath , 'Place_Link.svg')
         self.form.setWindowIcon(QtGui.QIcon( iconFile ))
-        self.form.setWindowTitle('Place linked Part')
+        self.form.setWindowTitle(App.Qt.translate("Asm4_placePart", 'Place linked Part'))
         self.drawUI()
 
         #save original AttachmentOffset of linked part
@@ -245,7 +237,7 @@
             # highlight in the 3D window the object we placed
             self.finish()
         else:
-            FCC.PrintWarning("Problem in selections\n")
+            FCC.PrintWarning(App.Qt.translate("Asm4_placePart", "Problem in selections\n"))
             return
 
 
@@ -270,7 +262,7 @@
             # see whether the ExpressionEngine field is filled
             if self.selectedObj.ExpressionEngine :
                 # if yes, then ask for confirmation
-                confirmed = Asm4.confirmBox('This command will release all attachments on '+msgName+' and set it to manual positioning in its current location.')
+                confirmed = Asm4.confirmBox(App.Qt.translate("Asm4_placePart", 'This command will release all attachments on '+msgName+' and set it to manual positioning in its current location.'))
                 # if not, then it's useless to bother the user
             else:
                 confirmed = True
@@ -280,10 +272,10 @@
                 # reset the assembly properties
                 Asm4.makeAsmProperties( self.selectedObj, reset=True )
                 # finish
-                FCC.PrintMessage("Part is now manually placed\n")
+                FCC.PrintMessage(App.Qt.translate("Asm4_placePart", "Part is now manually placed\n"))
                 self.finish()
             else:
-                FCC.PrintMessage("Part untouched\n")
+                FCC.PrintMessage(App.Qt.translate("Asm4_placePart", "Part untouched\n"))
                 self.finish()
 
 
@@ -298,7 +290,7 @@
     def Apply( self ):
         # get the instance to attach to:
         # it's either the top level assembly or a sister App::Link
-        if self.parentList.currentText() == 'Parent Assembly':
+        if self.parentList.currentText()=='Parent Assembly' or self.parentList.currentText()==App.Qt.translate("Asm4_placePart", 'Parent Assembly'):
             a_Link = 'Parent Assembly'
             a_Part = None
         elif self.parentList.currentIndex() > 1:
@@ -354,7 +346,7 @@
             self.rootAssembly.recompute(True)
             return True
         else:
-            FCC.PrintWarning("Problem in selections\n")
+            FCC.PrintWarning(App.Qt.translate("Asm4_placePart", "Problem in selections\n"))
             return False
 
 
@@ -366,7 +358,7 @@
         self.attLCStable = []
         # the current text in the combo-box is the link's name...
         # ... or it's 'Parent Assembly' then the parent is the 'Model' root App::Part
-        if self.parentList.currentText() == 'Parent Assembly':
+        if self.parentList.currentText()=='Parent Assembly' or self.parentList.currentText()==App.Qt.translate("Asm4_placePart", 'Parent Assembly'):
             parentName = 'Parent Assembly'
             parentPart = self.rootAssembly
             # we get the LCS directly in the root App::Part 'Model'
@@ -421,7 +413,7 @@
             a_LCS = self.attLCStable[ self.attLCSlist.currentRow() ]
             # get the part where the selected LCS is
             # parent assembly and sister part need a different treatment
-            if self.parentList.currentText() == 'Parent Assembly':
+            if self.parentList.currentText()=='Parent Assembly' or self.parentList.currentText()==App.Qt.translate("Asm4_placePart", 'Parent Assembly'):
                 a_LCStext = a_LCS.Name+'.'
             else:
                 a_Part = self.parentTable[ self.parentList.currentIndex() ].Name
@@ -578,7 +570,7 @@
         nbHash = expr.count('#')
         if nbHash==0:
             # linked part, sister part and assembly in the same document
-            if parent == 'Parent Assembly':
+            if parent=='Parent Assembly' or parent==App.Qt.translate("Asm4_placePart", 'Parent Assembly'):
                 # we're attached to an LCS in the parent assembly
                 # expr = LCS_in_the_assembly.Placement * AttachmentOffset * LCS_linkedPart.Placement ^ -1
                 ( attLCS,     separator, rest1 ) = expr.partition('.Placement * AttachmentOffset * ')
@@ -595,7 +587,7 @@
                 restFinal = rest3
         elif nbHash==1:
             # an external part is linked to the assembly
-            if parent == 'Parent Assembly':
+            if parent=='Parent Assembly' or parent==App.Qt.translate("Asm4_placePart", 'Parent Assembly'):
                 # we're attached to an LCS in the parent assembly
                 # expr = LCS_assembly.Placement * AttachmentOffset * LinkedPart#LCS.Placement ^ -1
                 ( attLCS,     separator, rest1 ) = expr.partition('.Placement * AttachmentOffset * ')
@@ -672,10 +664,10 @@
         self.parentTable = []
         self.parentList.clear()
         self.parentTable.append( [] )
-        self.parentList.addItem('Please select')
+        self.parentList.addItem(App.Qt.translate("Asm4_placePart", 'Please select'))
         self.parentTable.append( self.rootAssembly )
         parentIcon = self.rootAssembly.ViewObject.Icon
-        self.parentList.addItem( parentIcon, 'Parent Assembly', self.rootAssembly )
+        self.parentList.addItem( parentIcon, App.Qt.translate("Asm4_placePart", 'Parent Assembly'), self.rootAssembly )
         # set the old position values
         self.XtranslSpinBox.setValue(self.old_LinkPosition[0])
         self.YtranslSpinBox.setValue(self.old_LinkPosition[1])
@@ -693,14 +685,14 @@
         # Selected Link (the name as seen in the tree of the selected link)
         self.linkName = QtGui.QLineEdit()
         self.linkName.setReadOnly(True)
-        self.formLayout.addRow(QtGui.QLabel('Selected Link :'),self.linkName)
+        self.formLayout.addRow(QtGui.QLabel(App.Qt.translate("Asm4_placePart", 'Selected Link :')),self.linkName)
 
         # combobox showing all available App::Link
         self.parentList = QtGui.QComboBox()
         self.parentList.setMaximumWidth(300)
-        self.parentList.setToolTip('Choose the part in which the attachment\ncoordinate system is to be found')
+        self.parentList.setToolTip(App.Qt.translate("Asm4_placePart", 'Choose the part in which the attachment\ncoordinate system is to be found'))
         # the parent assembly is hardcoded, and made the first real element
-        self.formLayout.addRow(QtGui.QLabel('Attach to :'),self.parentList)
+        self.formLayout.addRow(QtGui.QLabel(App.Qt.translate("Asm4_placePart", 'Attach to :')),self.parentList)
         self.mainLayout.addLayout(self.formLayout)
 
         # with 2 columns
@@ -710,31 +702,31 @@
         # Part, left side
         #
         # the document containing the linked part
-        self.leftLayout.addWidget(QtGui.QLabel("Linked Part :"))
+        self.leftLayout.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_placePart", "Linked Part :")))
         self.linkedDoc = QtGui.QLineEdit()
         self.linkedDoc.setReadOnly(True)
         self.leftLayout.addWidget(self.linkedDoc)
 
         # The list of all LCS in the part is a QListWidget
-        self.leftLayout.addWidget(QtGui.QLabel("Select LCS in Part :"))
+        self.leftLayout.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_placePart", "Select LCS in Part :")))
         self.partLCSlist = QtGui.QListWidget(self.form)
         self.partLCSlist.setMinimumHeight(200)
-        self.partLCSlist.setToolTip('Select a coordinate system from the list')
+        self.partLCSlist.setToolTip(App.Qt.translate("Asm4_placePart", 'Select a coordinate system from the list'))
         self.leftLayout.addWidget(self.partLCSlist)
 
         # Assembly, Right side
         #
         # the document containing the linked object
-        self.rightLayout.addWidget(QtGui.QLabel("Parent Part :"))
+        self.rightLayout.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_placePart", "Parent Part :")))
         self.parentDoc = QtGui.QLineEdit()
         self.parentDoc.setReadOnly(True)
         self.rightLayout.addWidget(self.parentDoc)
         # The list of all attachment LCS in the assembly is a QListWidget
         # it is populated only when the parent combo-box is activated
-        self.rightLayout.addWidget(QtGui.QLabel("Select LCS in Parent :"))
+        self.rightLayout.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_placePart", "Select LCS in Parent :")))
         self.attLCSlist = QtGui.QListWidget(self.form)
         self.attLCSlist.setMinimumHeight(200)
-        self.attLCSlist.setToolTip('Select a coordinate system from the list')
+        self.attLCSlist.setToolTip(App.Qt.translate("Asm4_placePart", 'Select a coordinate system from the list'))
         self.rightLayout.addWidget(self.attLCSlist)
 
         # add the 2 columns
@@ -747,13 +739,13 @@
 
         # X Translation Value
         self.XoffsetLayout = QtGui.QHBoxLayout()
-        self.XtranslSpinBoxLabel = self.XoffsetLayout.addWidget(QtGui.QLabel("X Translation :"))
+        self.XtranslSpinBoxLabel = self.XoffsetLayout.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_placePart", "X Translation :")))
         self.XtranslSpinBox = Asm4.QUnitSpinBox()
         self.XtranslSpinBox.setRange(-999999.00, 999999.00)
         self.XtranslSpinBox.setDecimals(numberOfDecimals)
-        self.XtranslSpinBox.setToolTip("Translation along X axis")
-        self.RotXButton = QtGui.QPushButton('Rotate X +90°')
-        self.RotXButton.setToolTip("Rotate 90 deg around X axis")
+        self.XtranslSpinBox.setToolTip(App.Qt.translate("Asm4_placePart", "Translation along X axis"))
+        self.RotXButton = QtGui.QPushButton(App.Qt.translate("Asm4_placePart", 'Rotate X +90°'))
+        self.RotXButton.setToolTip(App.Qt.translate("Asm4_placePart", "Rotate 90 deg around X axis"))
         # add the QLDoubleSpinBox
         self.XoffsetLayout.addWidget(self.XtranslSpinBox)
         self.XoffsetLayout.addStretch()
@@ -762,13 +754,13 @@
 
         # Y Translation Value
         self.YoffsetLayout = QtGui.QHBoxLayout()
-        self.YtranslSpinBoxLabel = self.YoffsetLayout.addWidget(QtGui.QLabel("Y Translation :"))
+        self.YtranslSpinBoxLabel = self.YoffsetLayout.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_placePart", "Y Translation :")))
         self.YtranslSpinBox = Asm4.QUnitSpinBox()
         self.YtranslSpinBox.setRange(-999999.00, 999999.00)
         self.YtranslSpinBox.setDecimals(numberOfDecimals)
-        self.YtranslSpinBox.setToolTip("Translation along Y")
-        self.RotYButton = QtGui.QPushButton('Rotate Y +90°')
-        self.RotYButton.setToolTip("Rotate 90 deg around Y axis")
+        self.YtranslSpinBox.setToolTip(App.Qt.translate("Asm4_placePart", "Translation along Y"))
+        self.RotYButton = QtGui.QPushButton(App.Qt.translate("Asm4_placePart", 'Rotate Y +90°'))
+        self.RotYButton.setToolTip(App.Qt.translate("Asm4_placePart", "Rotate 90 deg around Y axis"))
         # add the QLDoubleSpinBox
         self.YoffsetLayout.addWidget(self.YtranslSpinBox)
         self.YoffsetLayout.addStretch()
@@ -777,13 +769,13 @@
 
         # Z Translation Value
         self.ZoffsetLayout = QtGui.QHBoxLayout()
-        self.ZtranslSpinBoxLabel = self.ZoffsetLayout.addWidget(QtGui.QLabel("Z Translation :"))
+        self.ZtranslSpinBoxLabel = self.ZoffsetLayout.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_placePart", "Z Translation :")))
         self.ZtranslSpinBox = Asm4.QUnitSpinBox()
         self.ZtranslSpinBox.setRange(-999999.00, 999999.00)
         self.ZtranslSpinBox.setDecimals(numberOfDecimals)
-        self.ZtranslSpinBox.setToolTip("Translation along Z:")
-        self.RotZButton = QtGui.QPushButton('Rotate Z +90°')
-        self.RotZButton.setToolTip("Rotate 90 deg around Z axis")
+        self.ZtranslSpinBox.setToolTip(App.Qt.translate("Asm4_placePart", "Translation along Z:"))
+        self.RotZButton = QtGui.QPushButton(App.Qt.translate("Asm4_placePart", 'Rotate Z +90°'))
+        self.RotZButton.setToolTip(App.Qt.translate("Asm4_placePart", "Rotate 90 deg around Z axis"))
         # add to the layout
         self.ZoffsetLayout.addWidget(self.ZtranslSpinBox)
         self.ZoffsetLayout.addStretch()
