--- Code/releaseAttachmentCmd.py	2025-09-11 10:39:02.509773972 +0800
+++ ../Assembly4-0.50.19/releaseAttachmentCmd.py	2025-07-04 15:18:48.192974653 +0800
@@ -39,8 +31,8 @@
 
 
     def GetResources(self):
-        return {"MenuText": "Release from Attachment",
-                "ToolTip": "Release an object from all attachments to any geometry",
+        return {"MenuText": App.Qt.translate("Asm4_releaseAttachment", "Release from Attachment"),
+                "ToolTip": App.Qt.translate("Asm4_releaseAttachment", "Release an object from all attachments to any geometry"),
                 "Pixmap" : os.path.join( Asm4.iconPath , 'Asm4_releaseAttachment.svg')
                 }
 
@@ -83,8 +75,8 @@
         objType  = selectedObj.TypeId
 
         # ask for confirmation before resetting everything
-        confirmText = 'This command will release all attachments on '+Asm4.labelName(selectedObj) \
-                    +' and set it to manual positioning in its current location.'
+        confirmText = App.Qt.translate("Asm4_releaseAttachment", 'This command will release all attachments on ')+Asm4.labelName(selectedObj) \
+                    +App.Qt.translate("Asm4_releaseAttachment", ' and set it to manual positioning in its current location.')
         if not Asm4.confirmBox(confirmText):
             # don't do anything
             return
