--- Code/selectionFilter.py	2025-09-11 10:39:37.527150408 +0800
+++ ../Assembly4-0.50.19/selectionFilter.py	2025-07-04 15:22:25.634447709 +0800
@@ -44,8 +35,8 @@
 """
 class selectionFilterClearCmd:
     def GetResources(self):
-        return {"MenuText": "Clear all selection filters",
-                "ToolTip": "Clear all selection filters",
+        return {"MenuText": App.Qt.translate("Asm4_selectionFilter", "Clear all selection filters"),
+                "ToolTip": App.Qt.translate("Asm4_selectionFilter", "Clear all selection filters"),
                 "Pixmap" : os.path.join( Asm4.iconPath , 'Asm4_SelectionAll.svg')
                 }
     def IsActive(self):
@@ -55,13 +46,13 @@
         Gui.Selection.removeSelectionGate()
         observerDisable()
         uncheckAll()
-        FCC.PrintMessage("All selection filters cleared\n")
+        FCC.PrintMessage(App.Qt.translate("Asm4_selectionFilter", "All selection filters cleared\n"))
 
 
 class selectionFilterVertexCmd:
     def GetResources(self):
-        return {"MenuText": "Select only Vertices",
-                "ToolTip": "Select only Vertices",
+        return {"MenuText": App.Qt.translate("Asm4_selectionFilter", "Select only Vertices"),
+                "ToolTip": App.Qt.translate("Asm4_selectionFilter", "Select only Vertices"),
                 "Pixmap" : os.path.join( Asm4.iconPath , 'Snap_Vertex.svg')
                 }
     def IsActive(self):
@@ -77,8 +68,8 @@
 
 class selectionFilterEdgeCmd:
     def GetResources(self):
-        return {"MenuText": "Select only Edges",
-                "ToolTip": "Select only Edges",
+        return {"MenuText": App.Qt.translate("Asm4_selectionFilter", "Select only Edges"),
+                "ToolTip": App.Qt.translate("Asm4_selectionFilter", "Select only Edges"),
                 "Pixmap" : os.path.join( Asm4.iconPath , 'Snap_Edge.svg')
                 }
     def IsActive(self):
@@ -94,8 +85,8 @@
 
 class selectionFilterFaceCmd:
     def GetResources(self):
-        return {"MenuText": "Select only Faces",
-                "ToolTip": "Select only Faces",
+        return {"MenuText": App.Qt.translate("Asm4_selectionFilter", "Select only Faces"),
+                "ToolTip": App.Qt.translate("Asm4_selectionFilter", "Select only Faces"),
                 "Pixmap" : os.path.join( Asm4.iconPath , 'Snap_Face.svg')
                 }
     def IsActive(self):
@@ -179,10 +170,10 @@
         Asm4_3DselObserver = None
 
     def GetResources(self):
-        return {"MenuText": "Enable/Disable 3D View selection mode",
-                "ToolTip": "Enable/Disable 3D View selection mode\n\n"    + \
+        return {"MenuText": App.Qt.translate("Asm4_selectionFilter", "Enable/Disable 3D View selection mode"),
+                "ToolTip": App.Qt.translate("Asm4_selectionFilter", "Enable/Disable 3D View selection mode\n\n"    + \
                 "Allows to select a Link object in the 3D view\n" + \
-                "window instead of the Model tree",
+                "window instead of the Model tree"),
                 "Pixmap" : os.path.join( Asm4.iconPath , 'Asm4_enableLinkSelection.svg')
                 }
 
@@ -242,7 +233,7 @@
     # add the listener, 0 forces to resolve the links
     Gui.Selection.addObserver(Asm4_3DselObserver, 0)
     setButton(3,True)
-    FCC.PrintMessage("Asm4 3D view selection mode is now ENABLED\n")
+    FCC.PrintMessage(App.Qt.translate("Asm4_selectionFilter", "Asm4 3D view selection mode is now ENABLED\n"))
 
 
 def observerDisable():
@@ -251,7 +242,7 @@
     setButton(3,False)
     # only print to Console if the Asm4_3DselObserver was there
     if Asm4_3DselObserver:
-        FCC.PrintMessage("Asm4 3D view selection mode is now DISABLED\n")
+        FCC.PrintMessage(App.Qt.translate("Asm4_selectionFilter", "Asm4 3D view selection mode is now DISABLED\n"))
     Asm4_3DselObserver = None
 
 
