--- Code/variantLinkCmd.py	2025-09-11 10:40:18.216749804 +0800
+++ ../Assembly4-0.50.19/variantLinkCmd.py	2025-07-04 15:31:10.335063260 +0800
@@ -41,11 +34,11 @@
         #pass
 
     def GetResources(self):
-        tooltip  = "EXPERIMENTAL !!!\n"
-        tooltip += "Create a variant link to a part\n"
-        tooltip += "Select a part containing a \"Variables\" property container"
+        tooltip  = App.Qt.translate("Asm4_variantLink", "EXPERIMENTAL !!!\n" \
+        + "Create a variant link to a part\n" \
+        + "Select a part containing a \"Variables\" property container")
         iconFile = 'Variant_Link.svg'
-        return {"MenuText" : "Create a variant Part", 
+        return {"MenuText" : App.Qt.translate("Asm4_variantLink", "Create a variant Part"),
                 "ToolTip"  :  tooltip, 
                 "Pixmap"   :  os.path.join( Asm4.iconPath, iconFile ) 
                 }
@@ -114,7 +107,7 @@
         '''
 
         if self.rootAssembly is None:
-            Asm4.warningBox( 'Please create an Assembly' )
+            Asm4.warningBox( App.Qt.translate("Asm4_variantLink", 'Please create an Assembly') )
             return
 
         # Search for all App::Parts having a "Variables" property container in all open documents
@@ -235,7 +228,7 @@
                 if self.rootAssembly == Asm4.getAssembly():
                     Gui.runCommand( 'Asm4_placeLink' )
             else:
-                Asm4.warningBox('The current document must be saved before inserting an external part')
+                Asm4.warningBox(App.Qt.translate("Asm4_variantLink", 'The current document must be saved before inserting an external part'))
                 return
 
         # if still open, close the dialog UI
@@ -299,7 +292,7 @@
         # make this dialog stay above the others, always visible
         self.UI.setWindowFlags( QtCore.Qt.WindowStaysOnTopHint )
         self.UI.setModal(False)
-        self.UI.setWindowTitle('Insert a variant of a Part')
+        self.UI.setWindowTitle(App.Qt.translate("Asm4_variantLink", 'Insert a variant of a Part'))
         self.UI.setWindowIcon( QtGui.QIcon( os.path.join( Asm4.iconPath , 'FreeCad.svg' ) ) )
         self.UI.setMinimumSize(400, 500)
         self.UI.resize(400,500)
@@ -312,18 +305,18 @@
         # Create a line that will contain the name of the link (in the tree)
         self.linkNameInput = QtGui.QLineEdit(self.UI)
         # Cancel button
-        self.cancelButton = QtGui.QPushButton('Cancel', self.UI)
+        self.cancelButton = QtGui.QPushButton(App.Qt.translate("Asm4_variantLink", 'Cancel'), self.UI)
         # Insert Link button
-        self.insertButton = QtGui.QPushButton('Insert', self.UI)
+        self.insertButton = QtGui.QPushButton(App.Qt.translate("Asm4_variantLink", 'Insert'), self.UI)
         self.insertButton.setDefault(True)
 
         # Place the widgets with layouts
         self.mainLayout = QtGui.QVBoxLayout(self.UI)
-        self.mainLayout.addWidget(QtGui.QLabel("Filter :"))
+        self.mainLayout.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_variantLink", "Filter :")))
         self.mainLayout.addWidget(self.filterPartList)
-        self.mainLayout.addWidget(QtGui.QLabel("Select Part to be inserted :"))
+        self.mainLayout.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_variantLink", "Select Part to be inserted :")))
         self.mainLayout.addWidget(self.partList)
-        self.mainLayout.addWidget(QtGui.QLabel("Name for the link :"))
+        self.mainLayout.addWidget(QtGui.QLabel(App.Qt.translate("Asm4_variantLink", "Name for the link :")))
         self.mainLayout.addWidget(self.linkNameInput)
         self.mainLayout.addWidget(QtGui.QLabel(' '))
         self.buttonsLayout = QtGui.QHBoxLayout()
